libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -199,19 +199,29 @@ void GCS_MAVLINK::send_battery_status(const uint8_t instance) const
     float temp;
     bool got_temperature = battery.get_temperature(temp, instance);
 
-    // ensure we always send a voltage estimate to the GCS, because not all battery monitors monitor individual cells
-    // as a work around for this we create a set of fake cells to be used if the backend doesn't provide direct monitoring
-    // the GCS can then recover the pack voltage by summing all non ignored cell values. Because this is looped we can
-    // report a pack up to 655.34 V with this method
-    AP_BattMonitor::cells fake_cells;
-    if (!battery.has_cell_voltages(instance)) {
+    // prepare array of individual cell voltage
+    uint16_t cell_volts[MAVLINK_MSG_BATTERY_STATUS_FIELD_VOLTAGES_LEN];
+    if (battery.has_cell_voltages(instance)) {
+        const AP_BattMonitor::cells& batt_cells = battery.get_cell_voltages(instance);
+        for (uint8_t i = 0; i < ARRAY_SIZE(batt_cells.cells); i++) {
+            if (i < MAVLINK_MSG_BATTERY_STATUS_FIELD_VOLTAGES_LEN) {
```

Comment:
Don't copy the first 8 cells via the for loop, just do a memcpy, which should be more efficent.

Reply:
Txs.. I'm going to give this suggestion a miss if that's OK.  You're right that it's slightly faster but it is such a small performance difference and I just think it's not worth the risk of getting it wrong.  I suspect you disagree so I won't stand in your way if you want to change it but I just know that I'll mess it up.